File: Forms/RunCommand.vb
Option Strict On
Option Explicit On

Public Class RunCommand
    Private Sub uiButtonOK_Click(ByVal sender As Object, ByVal e As EventArgs) Handles uiButtonOK.Click
        Dim lProcessInfo As New ProcessStartInfo() With
            {
                .FileName = uiTextBoxCommand.Text,
                .UseShellExecute = True,
                .ErrorDialog = True,
                .ErrorDialogParentHandle = Handle
            }

        Try
            Process.Start(lProcessInfo)
            Close()
        Catch ex As Exception
        End Try
    End Sub

    Private Sub uiButtonCancel_Click(ByVal sender As Object, ByVal e As EventArgs) Handles uiButtonCancel.Click
        DialogResult = DialogResult.Cancel
        Close()
    End Sub

    Private Sub uiComboBoxCommand_KeyUp(ByVal sender As Object, ByVal e As KeyEventArgs) Handles uiTextBoxCommand.KeyUp
        Select Case e.KeyCode
            Case Keys.Enter, Keys.Return
                e.Handled = True
                e.SuppressKeyPress = True
                uiButtonOK.PerformClick()
        End Select
    End Sub

    Private Sub uiButtonBrowse_Click(ByVal sender As Object, ByVal e As EventArgs) Handles uiButtonBrowse.Click
        If uiOpenFileDialog.ShowDialog = Windows.Forms.DialogResult.OK Then
            uiTextBoxCommand.Text = uiOpenFileDialog.FileName
        End If
    End Sub
End Class

File: QuickRun/Forms/RunCommand.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Linq;
using System.Windows.Forms;

namespace QuickRun.Forms
{
    public partial class RunCommand : Form
    {
        public RunCommand()
        {
            InitializeComponent();

            //Initialize event handlers
            uiButtonOK.Click += uiButtonOK_Click;
            uiButtonCancel.Click += uiButtonCancel_Click;
            uiButtonBrowse.Click += uiButtonBrowse_Click;
        }

        private void uiButtonOK_Click(object sender, EventArgs e)
        {
            ProcessStartInfo lProcessInfo = new ProcessStartInfo()
               {
                   FileName = uiTextBoxCommand.Text,
                   UseShellExecute = true,
                   ErrorDialog = true,
                   ErrorDialogParentHandle = Handle
               };

            try
            {
                Process.Start(lProcessInfo);
                Close();
            }
            catch (Exception ex)
            { }
        }

        private void uiButtonCancel_Click(object sender, EventArgs e)
        {
            DialogResult = DialogResult.Cancel;
            Close();
        }

        private void uiButtonBrowse_Click(object sender, EventArgs e)
        {
            if (uiOpenFileDialog.ShowDialog() == System.Windows.Forms.DialogResult.OK)
            {
                uiTextBoxCommand.Text = uiOpenFileDialog.FileName;
            }
        }
    }
}
